src/services/blockchain/ethereum/ethereum-blockchain-service.ts

```diff
@@ -89,14 +95,24 @@ export default class EthereumBlockchainService implements BlockchainService {
       txData.gasLimit = BigNumber.from(this.config.blockchain.connectors.ethereum.gasLimit)
       logger.debug('Overriding Gas limit: ' + txData.gasLimit.toString())
     } else {
-      const gasPriceEstimate = await this.wallet.provider.getGasPrice() // in wei
+      const feeData = await this.wallet.provider.getFeeData()
       // Add extra to gas price for each subsequent attempt
-      txData.gasPrice = EthereumBlockchainService.increaseGasPricePerAttempt(
-        gasPriceEstimate,
+      const prevMaxPriorityFeePerGas = BigNumber.from(txData.maxPriorityFeePerGas || 0)
+      const nextMaxPriorityFeePerGas = EthereumBlockchainService.increaseGasPricePerAttempt(
+        feeData,
         attempt,
-        txData.gasPrice
+        prevMaxPriorityFeePerGas
+      )
+      const difference = nextMaxPriorityFeePerGas.sub(prevMaxPriorityFeePerGas)
+      if (feeData.maxPriorityFeePerGas) {
+        txData.maxFeePerGas = feeData.maxFeePerGas.add(difference)
```

Comment:
By definition, `maxFeePerGas = baseFee + maxPriorityFeePerGas`. If we increase `maxPriorityFee`, we should make sure the equality still stands. If not adjusting for the difference we might end up `maxFeePerGas` being less than `baseFee + maxPriorityFeePerGas` (with maxPriorityFeePerGas being fixed). So this leads to `baseFee` being set too low, which reduces the chance of a tx being mined regardless of our `maxPriority` increase.

Reply:
Can we move this logic into `increaseGasPricePerAttempt` then?  Feels like half the logic happens there and half of it happens here.  Makes it confusing to follow

Reply:
or maybe `increaseGasPricePerAttempt` should just return the difference (and possibly be renamed), then it could be added to both `maxPriorityFeePerGas` and `maxFeePerGas` together here?  I just want to make sure it's easy to understand what this function is doing and how to use it correctly